src/db/pool.cr

```diff
@@ -47,6 +57,49 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T, delay : Float64)
+      sleep delay
+
+      sync do
+        connection_idle = @idle.includes? connection
+        return unless connection_idle # let the next reaping attempt get it
+
+        delete connection
+      end
+
+      connection.health_check! if connection.responds_to?(:health_check!)
+
+      sync do
+        @total << connection
+        @idle << connection
+      end
+    rescue exc : DB::ConnectionLost
+      {% if @top_level.has_constant?("Raven") %}
```

Comment:
leaving in for now to see how long a connection is alive for before a reaper w/frequency 60.0 detects it on Hiro

Reply:
Still planning on removing this before a final PR, but so far, haven't detected an error on Hiro